core/src/test/scala/org/apache/spark/shuffle/hash/HashShuffleReaderSuite.scala

```diff
@@ -117,9 +117,13 @@ class HashShuffleReaderSuite extends SparkFunSuite with LocalSparkContext {
     val mapOutputTracker = mock(classOf[MapOutputTracker])
     // Test a scenario where all data is local, just to avoid creating a bunch of additional mocks
     // for the code to read data over the network.
-    val statuses: Array[(BlockManagerId, Long)] =
-      Array.fill(numMaps)((localBlockManagerId, byteOutputStream.size().toLong))
-    when(mapOutputTracker.getServerStatuses(shuffleId, reduceId)).thenReturn(statuses)
+    var mapId = 0
```

Comment:
Can you to `(0 to numMaps).map{}` to eliminate the need for this mutable variable?